src/huggingface_hub/hf_api.py

```diff
@@ -255,14 +255,16 @@ class BlobSecurityInfo(TypedDict, total=False):
     av_scan: Optional[Dict]
     pickle_import_scan: Optional[Dict]
 
-
+@dataclass
 class TransformersInfo(TypedDict, total=False):
```

Comment:
```suggestion
@dataclass
class TransformersInfo(dict):
```

The class has to inherit from `dict` :)